### PR TITLE
Perform noncontroversal changes

### PR DESCRIPTION
**Pull request by Kulasko**
This PR should implement all noncontroversial changes listed by me in #1. One exception is the documentation, because this is your solution and I don't want to take away your chance to describe things the way you want.

### CODE REVIEW

**Comment by Kulasko** [comment]
It seems my solution wasn't present in this branch, so its commits slipped in too, my bad.

File: PrimeRust/solution_1/src/main.rs [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
+    time::{Duration, Instant},
 };
-use std::{thread, time::{Duration, Instant}};
 use structopt::StructOpt;
 
 pub mod primes {
     use std::{collections::HashMap, time::Duration, usize};
 
+    /// Shorthand for the `u8` bit count to avoid additional conversions.
+    const U8_BITS: usize = u8::BITS as usize;
```
Comment (mike-barber) [inline PrimeRust/solution_1/src/main.rs:16]: It looks like this was just added to stable in 1.53 -- and that reminds me of why I had it hard coded before :) 

We'll need to bump the version of Rust in the Docker file to 1.53 so it builds successfully.

**Review by mike-barber — CHANGES_REQUESTED** [review]
This looks really good -- we just need the bump to 1.53 in the Dockerfile.

Performance is almost identical, so luckily no surprises there :) 

Are you sure you want to include `solution_5` in this? You've already got a PR open for it, and this could potentially cause a bit of confusion. 

EDIT: Sorry I just saw your previous comment. What would you like to do? :)

File: PrimeRust/solution_1/src/main.rs (hunk 1 above)
Comment (Kulasko) [reply PrimeRust/solution_1/src/main.rs:16]: You are right - I forgot all about that, good catch.

Well, I don't really know how to attribute commits to different PRs after the fact - should I refork my branch from the main repo and reapply the commits from there?

Comment (mike-barber) [reply PrimeRust/solution_1/src/main.rs:16]: There's a way to cherry pick commits, but I don't know how to do it either :)

I think you should have permissions to push to a branch in this repo as you're a contributor. The easiest might just be to clone this repo, branch from `rust-idiomatic-improvements`, copy in the main.rs from here, etc. 

Going from the main repo should be fine too if that's easier :)

**Comment by Kulasko** [comment]
I close this in favor of #3.

**Pull request closed without merging**